libgiza/error.py

```diff
@@ -0,0 +1,239 @@
+"""
+Provides a simple, lightweight object used to capture data about an error
+encountered while checking or validating a resource.
+"""
+import logging
+import multiprocessing
+import sys
+import threading
+import traceback
+
+import libgiza.config
+
+logger = logging.getLogger('libgiza.error')
+
+if sys.version_info >= (3, 0):
+    basestring = str
+
+
+class Error(object):
+    """
+    Collects information about, and represents an error encountered. Use in
+    situations where you want to track errors, but don't want to raise
+    exceptions (e.g. for continue on error situations.)
+    """
+
+    def __init__(self, message="generic error", include_trace=True, fatal=True):
```

Comment:
"generic error" could be pulled out and made a constant, just to make a future typo more detectable.